### ISSUE TITLE
Client Feedback: Oral History and Web Annotation Integration

### ISSUE THREAD

**Issue by kstapelfeldt** [digitalutsc/islandora_web_annotations#188]
As a user, I would like to see annotation information appear in the same area as transcript information, because I find this less confusing. I would also like for the annotation tab to appear before (in front of) the transcription tab. 

Note: configuring the ordering of tabs would be useful as not all users will have this use case.

**Comment by kstapelfeldt** [comment]
Next step: UI sketch

**Comment by kstapelfeldt** [comment]
![oral-history-annotation-tab-user](https://cloud.githubusercontent.com/assets/467898/26552062/8678adb4-4453-11e7-9063-7095cd45ab35.png)

**Comment by MarcusBarnes** [comment]
Would we want to have the ability to toggle the annotation column like the speaker names?  Also, how would we want to deal with annotations that don't have start and end times within a given cue?  Thank you @kstapelfeldt .

**Comment by kstapelfeldt** [comment]
I don't think we have a use case for toggling the annotation column. Ideally, annotations do not rely on existing cues and annotation time cues are taken from the time code of the actual annotation object.

**Comment by kstapelfeldt** [comment]
When a user clicks a play button, the annotation would be highlighted the same way cues are highlighted. A user with appropriate permissions would edit the annotation using the existing affordance in the Web Annotation Solution Pack. Annotation content could appear or not above as is easiest to implement. If the decision should be made right away about whether or not content in the annotation appears both atop and below the vide, let me know and I will approach Chris and engage others in a wider discussion of the desired behaviour. 

![oral-history-end-user-highlighted-annotation](https://cloud.githubusercontent.com/assets/467898/26552466/40fe2780-4455-11e7-93c1-d8289a8cdad4.png)

**Comment by kstapelfeldt** [comment]
**Question of Overlapping Annotations:** I think a user would expect that the Annotations would appear listed by start time, and in the case of identical start times, could adopt some arbitrary or consistent order such as date annotation was created. 

**Question of Highlighting annotations during video play:** One question is how one would ideally experience highlighting for playing annotations in the case where the video is allowed to run. I believe the answer would be that the annotations would not highlight automatically as the video plays (and as the cue function for transcript works currently). The highlighting would only appear when a user actually clicks the play button.   I think it will be confusing if there are five cues highlighted at once though (would look jumpy)

**Configuration** Somewhere in Web Annotation Configuration, users would decide whether or not they want Oral History annotations to appear consolidated with the transcript viewer, and which tab would be open by default (Transcript or Annotation). Users should be able to use the default view or no in-text copy of the annotations if they wish.

**Comment by Natkeeran** [comment]
This ticket is related to : https://github.com/digitalutsc/islandora_web_annotations/issues/199

To add annotation view block to into the annotation tab: https://github.com/digitalutsc/islandora_solution_pack_oralhistories/tree/annotations_in_a_tab

**Comment by kstapelfeldt** [comment]
We need to dig in and explore/scope this feature.

**Comment by Natkeeran** [comment]
Depends on: https://github.com/digitalutsc/islandora_web_annotations/issues/199

* With respect to overlapping annotations, they can be ordered by creation time.  For now an edge case.  

* "The highlighting would only appear when a user actually clicks the play button." - We can indicate which play button was clicked with some effect in the block.

We can convert the transcript in oral history into a tab, and use https://www.drupal.org/project/block_tab or similar module to display as tabs.  This avoids the configuration part noted above as the user should have the ability to move the block up and down to set default behaviour.
